lib/vehicle.rb

```diff
@@ -6,13 +6,16 @@ class Vehicle
               :make,
               :model,
               :engine
+  attr_accessor :registration_date, :plate_type
 
   def initialize(vehicle_details)
-    @vin = vehicle_details[:vin]
+    @vin = vehicle_details[:vin] 
     @year = vehicle_details[:year]
     @make = vehicle_details[:make]
     @model = vehicle_details[:model]
     @engine = vehicle_details[:engine]
+    @registration_date = registration_date
+    @plate_type = plate_type
```

Comment:
Technically you're setting @registration_date and @plate_type to nil here since you don't have a variable named registration_date or plate_type.